random-menu-generator.rb

```diff
@@ -0,0 +1,784 @@
+#                   *  Dr. Vonn Jerry XLII edition  *
+#                   *                               *
+#                   * * * * * * * * * * * * * * * * *
+#
+# Notes:
+#
+# 1. I really enjoyed the spoken element that Victoriya added to her calculator
+#    project, so I decided to add a spoken word feature to this program. This
+#    feature for mac users actually seems pretty useful, because you could start
+#    the program and then walk over to various cabinets to check for ingredients
+#    while the program is reeling off the resultant list.
+#
+#    *  It won't work for non mac users, and I wasn't sure how to handle that.
+#
+#    *  And because I felt I was writing this for future me, I probably did not
+#       create the best user experience for users other than me. I think this
+#       program does a pretty poor job of telling the user what it is doing or
+#       what things it knows how to do.
+#
+# 2. Also, I think this will be a much more useful thingy after we learn how to
+#    interact with outside files. The database stuff could be stored in a text
+#    file, which would mean items added by the user could be remembered.
+#
+#    *  On a related note, if you see something like !W !T !R it's probably a
+#       note for my future self for the next time I work on this (in case I
+#       forget some of the work or tests I wanted to do.
+#
+# Enjoy!
+#    --Jeri / @drvonnjerryxlii
+
+
+# Expand your solution to ensure that no descriptive term in a menu item is
+# ever {"} repeated. SORT OF !Q uniq in list or uniq in soft softish eggs?
+
+
+#------------------------- BEGIN WEIRD GLOBAL THINGS ---------------------------
+
+$voice = true # speak until user says not to
+$individual = true # aka not combination
+$what_kind_of_ideas_desired = nil # what can I say? I don't have any ideas yet.
+$how_many_ideas_desired = 0 # and I don't want any, either!
+$user_wants_too_many = false # zero just doesn't know how to be too many.
+
+#-------------------------- END WEIRD GLOBAL THINGS ----------------------------
+#--------------------- BEGIN OUTPUT TEXT TO USER THINGS ------------------------
+
+# outputs provided string of text to terminal & possibly to speakers as well via
+# %x interaction with outside terminal environment. !W discovered the hard way
+# that say does not take some punctuation well: & ' " \ (
+def speak(string_to_speak)
+  puts "#{string_to_speak}"
+
+  if $voice
+    good_voices = ["Alex", "Bad", "Good", "Vicki", "Zarvox"]
+    random_voice = good_voices[(rand * good_voices.length)]
+    %x{say -v #{random_voice} #{string_to_speak}}
+  end
+end
+
+
+
+
+# checks whether the voice should be toggle from true to false or vice versa by
+# looking for specific keywords in a provided string.
+def check_reset_voice(user_input)
+  voice_triggers = %w{speak word spoke say speech talk tell said}
+
+  voice_triggers.each do |voice_trigger|
+    if user_input.downcase.include? voice_trigger
```

Comment:
I'm noticing this pattern a lot in this code, where you loop through an array to check if there's a match in an input string. You should check out [regular expressions](http://ruby-doc.org/core-2.2.0/Regexp.html). There are probably some methods in that class that could help you make these checks a little more concise and efficient.